### ISSUE TITLE
forcefield.py tests are lazy

### ISSUE THREAD

**Issue by arm61** [arm61/pylj#43]
Improve the forcefield tests such that:
- `dr` is an array 
- different values of `constants` are being used

**Comment by arm61** [comment]
@sansona has made some improvements with #45, #47 and #48. Does we have good coverage now of edge cases?

**Comment by arm61** [comment]
oppps, didn't mean to close yet

**Comment by bjmorgan** [comment]
Yes. `forcefield.py` no longer exists: 2.0 replaced it with `potentials.py`, where each potential is a class implementing a `PairPotential` interface, so the tests were written fresh rather than extended.

Against the two points in the opening comment:

- `dr` is an array throughout. `energies` and `forces` take an array of separations and return an array of the same shape, and the tests pass arrays in twenty-one places, including the grids used to check the Buckingham barrier and the square well's three regions.
- Parameters are varied rather than fixed. Each constructor is exercised across zero, negative, infinite and not-a-number values for every parameter it takes, and the physics is checked at known points: the Lennard-Jones zero crossing and well minimum, the Buckingham collapse at short range, and the square well at each of its walls.

The file has 33 tests across five classes, and `pylj/potentials.py` is at 100 percent line and branch coverage.

Closing as done.
